crates/ruff_db/src/files.rs

```diff
@@ -424,6 +424,13 @@ impl File {
     pub fn exists(self, db: &dyn Db) -> bool {
         self.status(db) == FileStatus::Exists
     }
+
+    /// Returns `true` if the file should be analyzed as a type stub.
+    pub fn is_stub(self, db: &dyn Db) -> bool {
+        self.path(db)
+            .extension()
+            .is_some_and(|extension| extension == "pyi")
```

Comment:
I'd use `ruff_python_ast::PySourceType` here:

```suggestion
            .is_some_and(|extension| PySourceType::from_extension(extension).is_stub())
```

if nothing else, for a bit more symmetry with the `is_notebook` function here:

https://github.com/astral-sh/ruff/blob/cfafaa7637ef3f568d62d39654e0498f52a07b77/crates/ruff_db/src/source.rs#L54-L66